cli/server.go

```diff
@@ -1512,3 +1517,68 @@ func redirectHTTPToAccessURL(handler http.Handler, accessURL *url.URL) http.Hand
 func isLocalhost(host string) bool {
 	return host == "localhost" || host == "127.0.0.1" || host == "::1"
 }
+
+func makeLogger(cmd *cobra.Command, cfg *codersdk.DeploymentConfig) (slog.Logger, func(), error) {
+	var (
+		sinks   = []slog.Sink{}
+		closers = []func() error{}
+	)
+
+	if cfg.Logging.Human.Value != "" {
+		if cfg.Logging.Human.Value == "/dev/stderr" {
+			sinks = append(sinks, sloghuman.Sink(cmd.ErrOrStderr()))
+		} else {
+			fi, err := os.OpenFile(cfg.Logging.Human.Value, os.O_WRONLY|os.O_CREATE|os.O_APPEND, 0644)
+			if err != nil {
+				return slog.Logger{}, nil, xerrors.Errorf("open human log %q: %w", cfg.Logging.Human.Value, err)
+			}
+			closers = append(closers, fi.Close)
+			sinks = append(sinks, sloghuman.Sink(fi))
+		}
+	}
+
+	if cfg.Logging.JSON.Value != "" {
+		if cfg.Logging.JSON.Value == "/dev/stderr" {
+			sinks = append(sinks, slogjson.Sink(cmd.ErrOrStderr()))
+		} else {
+			fi, err := os.OpenFile(cfg.Logging.JSON.Value, os.O_WRONLY|os.O_CREATE|os.O_APPEND, 0644)
+			if err != nil {
+				return slog.Logger{}, nil, xerrors.Errorf("open json log %q: %w", cfg.Logging.JSON.Value, err)
+			}
+			closers = append(closers, fi.Close)
+			sinks = append(sinks, slogjson.Sink(fi))
+		}
```

Comment:
This whole routine could be made into a little helper function (even just a function stored in a variable at the top of this function would be good) that returns an io.Writer